src/coreclr/debug/daccess/dacdbiimpl.cpp

```diff
@@ -7535,6 +7535,31 @@ HRESULT DacDbiInterfaceImpl::EnableGCNotificationEvents(BOOL fEnable)
     return hr;
 }
 
+HRESULT DacDbiInterfaceImpl::ReadData(TADDR pRemoteBuf, DWORD size, BYTE *pLocalBuf)
+{
+    DD_ENTER_MAY_THROW
+
+    HRESULT hr = S_OK;
+    EX_TRY
+    {
+        ULONG32 cbRead = 0;
+
+        HRESULT hr = m_pTarget->ReadVirtual(pRemoteBuf, pLocalBuf, size, &cbRead);
+
+        if (FAILED(hr))
+        {
+            ThrowHR(CORDBG_E_READVIRTUAL_FAILURE);
```

Comment:
Why are you throwing an exception here and then catching below? When you could just `return CORDBG_E_READVIRTUAL_FAILURE`?